concurrency/SortXDFs.java

```diff
@@ -0,0 +1,994 @@
+            } else {
+                familyIdDataOut.writeByte((byte) veFamilyId.length);
+                familyIdDataOut.writeBytes(veFamilyId.bytes, veFamilyId.offset, veFamilyId.length);
+            }
+            return familyIdOut.toByteArray();
+        }
+
+        // XDF has one document per line:
+        private int findEndFragment(byte[] bytes) {
+            for (int i = bytes.length - 1; i >= 0; i--) {
+                if (bytes[i] == '\n') {
+                    return i + 1;
+                }
+            }
+
+            return -1;
+        }
+
+        private void writeField(XDFParser parser, boolean isItem, String docId, DataOutput out,
+                                String fieldName, String value) throws IOException {
+
+            Integer fieldId = fieldIds.get(fieldName);
+
+            if (fieldName.startsWith("proffer/")) {
+                // TODO:
+                fieldId = getFieldId(fieldName);
+                writeString(out, fieldId, value);
+                return;
+            }
+            if (fieldName.contains("fitment")) {
+                // TODO:
+                fieldId = getFieldId(fieldName);
+                writeString(out, fieldId, value);
+                return;
+            }
+
+            switch (fieldName) {
+                // the <build-index name="single-index"> skips these:
+                case "author-strip":
+                case "brandtextbin-jp":
+                case "ean":
+                case "si-strip-common-body":
+                case "subject-strip":
+                case "title-strip":
+                case "upc":
+                case "sku":
+                    break;
+                // these seem special cased?
+                case "a9-doc-version":
+                case "_language":
+                case "service-number":
+                case "service-id":
+                    break;
+                case "ve-family-id":
+                    value = fixVEFamilyID(value);
+                    writeString(out, fieldId, value);
+                    break;
+                case "ve-family-id2":
+                    value = fixVEFamilyID(value);
+                    writeString(out, fieldId, value);
+                    break;
+                case "ve-family-id3":
+                    value = fixVEFamilyID(value);
+                    writeString(out, fieldId, value);
+                    break;
+                case "single-index-body":
+                    String attrSource = parser.getAttrSource();
+                    int sourceFieldId = getFieldId(attrSource);
+                    writeSourceString(out, fieldId, sourceFieldId, value);
+                    break;
+                case "is_hdr_video":
+                    long longValue;
+                    try {
+                        longValue = Long.parseLong(value);
+                    } catch (NumberFormatException nfe) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)